casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -106,11 +52,11 @@ object Validate {
       _ <- EitherT(Validate.blockNumber(block))
       // Deploys validation
       _ <- EitherT.liftF(Span[F].mark("before-deploys-shard-identifier-validation"))
-      _ <- EitherT(Validate.deploysShardIdentifier(block, shardId))
+      _ = BlockValidationLogic.deploysShardIdentifier(block, shardId)
       _ <- EitherT.liftF(Span[F].mark("before-future-transaction-validation"))
-      _ <- EitherT(Validate.futureTransaction(block))
+      _ = BlockValidationLogic.futureTransaction(block)
       _ <- EitherT.liftF(Span[F].mark("before-transaction-expired-validation"))
-      _ <- EitherT(Validate.transactionExpiration(block, expirationThreshold))
+      _ = BlockValidationLogic.transactionExpiration(block, expirationThreshold)
```

Comment:
These expression are not doing anything now. You need to convert them to EitherT to have short-circuit behaviour if they are failed.

For now you can write some simplified version (just as a prototype) of the test for this function with Mockito lib. It can be just the proof that EitherT from Boolean works correctly.

Reply:
Yes, you are right. I missed it :roll_eyes: I [fixed it and added some simple tests](https://github.com/rchain/rchain/pull/3793/commits/919477534003ec025287eea3cb68c1fe963aec2f).